src/ServiceControl.Monitoring/Http/Diagrams/TimingsAggregator.cs

```diff
@@ -21,22 +25,35 @@ public IEnumerable<MonitoredEndpoint> AggregateIntoLogicalEndpoints()
 
             var processingTimes = processingTimeStore.GetTimings(now);
             var criticalTimes = criticalTimeStore.GetTimings(now);
+            var queueLengths = queueLengthDataStore.GetQueueLengths(now);
 
-            var endpointNames = processingTimes.Concat(criticalTimes).Select(t => t.Id.EndpointName).Distinct();
+            var endpointNames = processingTimes.Concat(criticalTimes).Select(t => t.Id.EndpointName).Concat(queueLengths.Keys).Distinct();
 
             foreach (var endpointName in endpointNames)
             {
                 var instanceProcessingTime = processingTimes.Where(pt => pt.Id.EndpointName == endpointName).ToList();
                 var instanceCriticalTime = criticalTimes.Where(ct => ct.Id.EndpointName == endpointName).ToList();
+                var queueLength = queueLengths.FirstOrDefault(ql => ql.Key == endpointName);
```

Comment:
I think that we need more data to answer this one. My gut feeling is that we should be ok with the current approach. The reason is that number number `ServicePulse` instances being used at the same time will most likely never exceed more than a couple and number of endpoints in range of 100s.